File: container_most_water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        m_area = 0
        while l < r:
            shorter = height[r] if height[r] < height[l] else height[l]
            area = shorter * (r - l)
            m_area = area if area > m_area else m_area
            if height[r] < height[l]: 
                r -= 1
            else:
                l += 1
                
        return m_area
        

File: ContainerMostWater.cs
public class Solution {
    public int MaxArea(int[] height) {
        int l = 0;
        int r = height.Length - 1;
        int MinArea = 0;
        while (l < r ) {
            int shorter = height[r] < height[l] ? height[r] : height[l];
            int area = shorter * (r - l);
            MinArea = area > MinArea ? area : MinArea;
            if (height[r] < height[l]) r -= 1;
            else l+= 1;
        }
        return MinArea;
    }
}